atequiz/index.ts

```diff
@@ -26,6 +26,13 @@ export interface AteQuizResult {
   hintIndex: number | null;
 }
 
+type AteQuizMode = 'normal' | 'solo';
+
+export interface AteQuizStartOption {
+  mode: AteQuizMode;
+  player?: string;
+}
```

Comment:
[may]

```suggestion
export interface NormalAteQuizStartOption {
  mode: 'normal';
}

export interface SoloAteQuizStartOption {
  mode: 'solo';
  player: string;
}

export type AteQuizStartOption = NormalAteQuizStartOption | SoloAteQuizStartOption;
```

のほうがうれしそう

Reply:
これってどういう嬉しさがあるんですか？

Reply:
異なるオブジェクト型のunionにすることで、(modeプロパティをチェックすることにより) 自動的に type guard することができるようになります。

例えば

```typescript
let option: AteQuizStartOption;
if (option.mode === 'normal') {
  console.log(option.player);
}
```

がエラーになります。

https://typescript-jp.gitbook.io/deep-dive/type-system/typeguard#riterarunotype-guard などを参照

Reply:
type guard えらすぎ？
ありがとうございます！

Reply:
文字列リテラルが型になるの知らんかったな